File: udemy/ds-and-a/DSAndA/LinkedLists/Node.cs
namespace DSAndA.LinkedLists
{
    public class Node<T>
    {
        private T data;
        public Node<T> NextNode { get; set; }

        public Node(T data)
        {
            this.data = data;
        }

        public override string ToString()
        {
            return this.data.ToString();
        }

        public T GetData()
        {
            return this.data;
        }

        public void OverwriteData(T data)
        {
            this.data = data;
        }
    }
}

File: udemy/ds-and-a/javascript-solutions/src/LinkedLists/Node.js
class Node {
    constructor(data) {
        this.data = data;
        this.nextNode = null;
    }

    get data() {
        return this._data;
    }

    get nextNode() {
        return this._nextNode;
    }

    set data(data) {
        this._data = data;
    }

    set nextNode(nextNode) {
        this._nextNode = nextNode;
    }
}

module.exports = Node;